packages/react/src/components/CheckboxGroup/CheckboxGroup.tsx

```diff
@@ -89,6 +96,44 @@ const CheckboxGroup: React.FC<CheckboxGroupProps> = ({
     ? cloneElement(candidate, { size: 'mini', kind: 'default' })
     : null;
 
+  const clonedChildren = Children.map(children, (child) => {
+    if (
+      isValidElement<ComponentProps<typeof Checkbox>>(child) &&
+      child.type === Checkbox
+    ) {
+      const childProps: Partial<
+        Pick<ComponentProps<typeof Checkbox>, 'invalid' | 'readOnly' | 'warn'>
+      > = {};
+
+      if (
+        typeof invalid !== 'undefined' &&
+        typeof child.props.invalid === 'undefined'
+      ) {
+        childProps.invalid = invalid;
+      }
+
+      if (
+        typeof readOnly !== 'undefined' &&
+        typeof child.props.readOnly === 'undefined'
+      ) {
+        childProps.readOnly = readOnly;
+      }
+
+      if (
+        typeof warn !== 'undefined' &&
+        typeof child.props.warn === 'undefined'
+      ) {
+        childProps.warn = warn;
+      }
```

Comment:
Everything looks good, just a small improvement here




```suggestion
    const childProps = Object.assign(
  {},
  typeof invalid !== 'undefined' && typeof child.props.invalid === 'undefined' && { invalid },
  typeof readOnly !== 'undefined' && typeof child.props.readOnly === 'undefined' && { readOnly },
  typeof warn !== 'undefined' && typeof child.props.warn === 'undefined' && { warn }
);
```

Reply:
> just a small improvement here

I don't see how that suggestion is an improvement. That code would result in a type error and has worse performance characteristics. I'm willing to update the implementation to be more concise, but I'd prefer a different approach than the one suggested.